samples/boards/nrf/nrfx/Kconfig

```diff
@@ -2,10 +2,10 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config NRFX_DPPI
-	default $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_DPPIC))
+	default HAS_HW_NRF_DPPIC
```

Comment:
The actual problem here is that the `DT_COMPAT_NORDIC_NRF_DPPIC` symbol is not defined yet, since "Kconfig.zephyr" is sourced at the end of this file, and in my opinion this should be corrected by moving the `source` instruction.

`HAS_HW_NRF_DPPIC` evaluates to `$(dt_compat_enabled,...)` so it's not entirely correct here - the thing is that the nrfx_dppi driver is to be enabled when DPPI is available on a given SoC, not only when the corresponding node is enabled in devicetree (it is always enabled now, so currently the effect is the same). It's just a detail, but I think it can be misleading.

Reply:
move `source Kconfig.zephyr` to the top.